lnwallet/commitment.go

```diff
@@ -303,10 +304,30 @@ func CommitScriptToRemote(chanType channeldb.ChannelType, initiator bool,
 			WitnessScript: script,
 		}, 1, nil
 
+	// For taproot channels, we'll use a slightly different format, where
+	// the top-level key is the combined funding key (w/o the bip 86
+	// tweak), with the sole tap leaf enforcing the 1 CSV delay.
+	case chanType.IsTaproot():
```

Comment:
pre-existing with the original set of cases, but is it not possible in future for channels with lease expirations to one day also be taproot channels? in which case the first case would trigger & the script would be wrong

Reply:
Correct, this PR series doesn't yet support the lease expiration channel variant. Plan was to do that in another pass, as we already gate the new channel type based on the channel type TLV in the `open_channel` message.